### DISCUSSION TITLE
Multiple Contracts vs One

### DISCUSSION THREAD

**Discussion by 20skcs** [smartcontractkit/full-blockchain-solidity-course-py#977] [Q&A]
Perhaps an obvious answer to this (and my apologies if it is) but I'm trying to better understand the best structure of some more complex contracts. That has brought me to the question of whether it is preferable/better practice to break a particular objective up into multiple contracts or code them together as one?

I don't have a specific example that I can detail, but by way of a simple explanation, say I wanted to combine the functionality of the FundMe.sol contract and the SimpleCollectible.sol in the following way:-

People wanting to support me can send eth to the contract and if they wish, mint an NFT (with characteristics already defined) as long as the pay the gas for the minting. Would that be better achieved as two contracts interacting with each other (e.g. a base contract with functionality of FundMe.sol which inherits functions from SimpleCollectible) or somehow coded as one? Or does it not really matter (assuming the maximum contract size is not exceeded)?

Again, my apologies if my lack of detailed knowledge in this space makes the way I am framing this question nonsensical. More just after some guiding thoughts on the topic if possible, thank you in advance.

**Comment by cromewar** [comment]
Hello @20skcs What an interesting topic, my answer in this case is: Depends, on the context, the contract situation or the client requirements. Let me go more in deep into this.

1. As you said there is a size limit for the contracts, so, that's your first thing to check out, if I combine this (one, two or more) contracts into just one, it'll have enough space?
2. Apart from that, I would say is more a matter of preference, a programming problem could be solved on a million different forms and depends a lot on how the coder (or software architect) is planning it. Of course there are ways more optimal than others to write code, but on the 95% scenarios it won't actually matter as long as the code works.
3. A key part here on blockchain is the `gas cost`, a better implemented code will lead to less gas costs which is preferably in almost most cases.

Regarding to your specific problem:

The FundMe contract is more related to funding indeed, so if your goal is to ask for payment on a certain token in order to be able to mint a NFT I would suggest to just use a `require` statement and declare the token as an interface. I've made a project of this you can see the contract code [here](https://github.com/cromewar/pomobots).
